src/toolkits/object_detection/object_detector.cpp

```diff
@@ -493,6 +493,22 @@ void object_detector::perform_predict(gl_sframe data,
   // For each anchor box, we have 4 bbox coords + 1 conf + one-hot class labels
   int num_outputs_per_anchor = 5 + static_cast<int>(class_labels.size());
   int num_output_channels = num_outputs_per_anchor * anchor_boxes().size();
+
+  float_array_map pred_config = get_prediction_config();
+  pred_config["num_classes"] =
+      shared_float_array::wrap(data_iter->class_labels().size());
```

Comment:
Let's move these into the `get_prediction_config` and `get_training_config` functions. These functions may need to take arguments for things like the number of classes. Per offline discussion, looks like `get_base_config` could be useful for the shared portion